src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaWrapper.cs

```diff
@@ -37,10 +45,32 @@ public class AWSLambdaWrapper
         /// <param name="tracerProvider">TracerProvider passed in.</param>
         /// <param name="lambdaHandler">Lambda handler function passed in.</param>
         /// <param name="input">Instance of input.</param>
+        /// <param name="parentContext">
+        /// The optional parent context <see cref="ActivityContext"/> used for Activity object creation.
+        /// If the parent context parametere is not provided then X-Ray propagation is used to extract the parent.
+        /// </param>
         /// <returns>Instance of output result.</returns>
-        public static TResult Trace<TInput, TResult>(TracerProvider tracerProvider, Func<TInput, TResult> lambdaHandler, TInput input)
+        public static TResult Trace<TInput, TResult>(
+            TracerProvider tracerProvider,
+            Func<TInput, TResult> lambdaHandler,
+            TInput input,
+            ActivityContext parentContext = default)
         {
-            return Intercept(tracerProvider, () => lambdaHandler(input));
```

Comment:
Why was the `Intercept` pattern changed? Couldn't that approach still be used to reduce code duplication between the overloads?